Most likely the start of a 2.0 refactor

JohnathonKoster opened the pull request:
This PR introduces a new `replace` method that can be used to specify a list of keys that should be overwritten entirely, even when performing a `mergeMany`.

In the following, the value of `some_key` will be entirely replaced with the incoming data, while all existing keys will use the merge behavior:

```php
<?php

Stillat\Proteus\Support\Facades\ConfigWriter;

ConfigWriter::replace(['some_key']->mergeMany('app', [
   'other_key' => [],
   'some_key' => []
]);
```

JohnathonKoster committed:
Adds simpler way to go into writeMany mode for specific keys

StyleCIBot committed:
Apply fixes from StyleCI

JohnathonKoster committed:
Merge pull request #20 from Stillat/analysis-7W2VgZ

Apply fixes from StyleCI

edalzell commented:
This works great @JohnathonKoster, don't forget to update the ConfigWriterContract

JohnathonKoster committed:
Make Things Simpler To Use

StyleCIBot committed:
Apply fixes from StyleCI

JohnathonKoster committed:
Merge pull request #22 from Stillat/analysis-YOQyjo

Apply fixes from StyleCI

JohnathonKoster committed:
Update ConfigWriterContract.php

JohnathonKoster committed:
Merge branch 'replace-while-merging' of https://github.com/Stillat/proteus into replace-while-merging

JohnathonKoster committed:
🍺

JohnathonKoster committed:
That sucked.

JohnathonKoster committed:
Fix that bug before Erin finds it

JohnathonKoster committed:
Let's ignore function calls by default (almost everyone calls this anyway)

JohnathonKoster committed:
Remove mergeMany from the wrapper

This just causes way too much confusion. This feature will still be available for those that need it in automated processes by accessing the ConfigUpdater directly.

JohnathonKoster committed:
Adds test to ensure trailing comments are preserved

JohnathonKoster committed:
Refactor tests to namespaces

JohnathonKoster committed:
Cleanup - ignore functions is now the default behavior

JohnathonKoster committed:
Add coverage for adding a new dynamic node

JohnathonKoster committed:
Update README.md

JohnathonKoster committed:
More sanity testing

JohnathonKoster committed:
Makes write smarter about inserts,  by default

JohnathonKoster committed:
Updates examples

The automatic dot notation is just too much of a pain to support in addition to all of the magic already going on

JohnathonKoster committed:
Cleanup 🧹

JohnathonKoster committed:
Provide option to get to previous behavior

edalzell commented:
@JohnathonKoster this works perfect for both of my scenarios, SHIP IT!
